include/mppic/parameters_handler.hpp

```diff
@@ -0,0 +1,138 @@
+// Copyright 2022 FastSense, Samsung Research
+
+#pragma once
+
+#include <vector>
+#include <functional>
+#include <type_traits>
+
+#include "rclcpp/rclcpp.hpp"
+#include "rclcpp_lifecycle/lifecycle_node.hpp"
+#include "nav2_util/node_utils.hpp"
+
+namespace mppi
+{
+
+enum class ParameterType
+{
+  Dynamic,
+  Static
```

Comment:
Unless you think you want to add static params, I'd just assume everything is a dynamic parameter and remove the enum / argument fields for it. If this was for `rclcpp` or `nav2_utils` I'd say this is good to be general, but if its just for this package, I think we can just assume everything is changable

Reply:
Maybe some parameters i don't want to be dynamically changeable in future/may be now, i didn't think about this a lot yet. It's easy feature, so why don't just have it.

Reply:
Mostly because the user code always specifies `DYNAMIC` when setting up the parameters. Maybe just set the default type to dynamic so that all the locations in the code don't need to specify it unless explicitly static? That's the default behavior in ROS 2 as well - everything is dynamic unless otherwise set

Reply:
Done